src/data/fintype/basic.lean

```diff
@@ -1019,6 +1019,16 @@ fintype.of_surjective quotient.mk (λ x, quotient.induction_on x (λ x, ⟨x, rf
 instance finset.fintype [fintype α] : fintype (finset α) :=
 ⟨univ.powerset, λ x, finset.mem_powerset.2 (finset.subset_univ _)⟩
 
+private def equiv_inj_subtype (α β : Sort*) : {f : α → β // function.injective f} ≃ (α ↪ β) :=
```

Comment:
Why is this `private`? Really things shouldn't be `private` in mathlib unless in exceptional circumstances. It's saying that you're _certain_ no one will ever want to use it.

Reply:
name should probably be `subtype_injective_equiv_embedding`?

Reply:
If anything I think this could be helpful - maybe in `data/equiv/basic`?

Reply:
I suggested this be private because I can't think of any situation you would ever need it. It's only used here as glue to the fintype api. If `private` is scary, then maybe we could inline it into the proof where it's used?

Reply:
> I suggested this be private because I can't think of any situation you would ever need it.

I disagree - I think it's helpful to have equivalences like these comparing explicit subtypes to bundled functions, especially when working with chains of equivalences. Here, it doesn't even relate to fintypes specifically: I think it should be a public lemma and be added to the API for embeddings or equivalences.

Reply:
Yes, I agree this is a useful equivalence should be part of the API.

Reply:
I moved this to `logic/embedding` under the `equiv` namespace, and made it public!